Storage/ThirdParty/SwiftData.swift

```diff
@@ -93,17 +93,23 @@ public class SwiftData {
      * The real meat of all the execute methods. This is used internally to open and
      * close a database connection and run a block of code inside it.
      */
-    public func withConnection(flags: SwiftData.Flags, synchronous: Bool=true, cb: (db: SQLiteDBConnection) -> NSError?) -> NSError? {
-        let conn: SQLiteDBConnection?
+    func withConnection(flags: SwiftData.Flags, synchronous: Bool=true, cb: (db: SQLiteDBConnection) -> NSError?) -> NSError? {
+        let conn: ConcreteSQLiteDBConnection?
 
         if SwiftData.ReuseConnections {
             conn = getSharedConnection()
         } else {
             log.debug(">>> Creating non-shared SQLiteDBConnection for \(self.filename) on thread \(NSThread.currentThread()).")
-            conn = SQLiteDBConnection(filename: filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey)
+            conn = ConcreteSQLiteDBConnection(filename: filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey)
         }
 
         guard let connection = conn else {
+            // Run the callback with a fake failed connection.
+            // Yeah, that means we have an error return value but we still run the callback.
```

Comment:
Notably, callers tend not to check the error result of `withConnection`, because they're wrapping it with their own semantics.